### PR TITLE
openstack: add 4.2 serial canary job

### PR DESCRIPTION
**Pull request by dprince**

### CODE REVIEW

File: ci-operator/infra/openshift/release-controller/releases/release-ocp-4.2.json [hunk 1]
```diff
@@ -78,6 +78,10 @@
       "optional":true,
       "prowJob":{"name":"canary-openshift-ocp-installer-e2e-openstack-4.2"}
     },
+    "openstack":{
```
Comment (AlexNPavel) [inline ci-operator/infra/openshift/release-controller/releases/release-ocp-4.2.json]: This should be changed to `openstack-serial`

Comment (dprince) [reply ci-operator/infra/openshift/release-controller/releases/release-ocp-4.2.json]: Good catch. I just pushed an update to fix this.

**Comment by smarterclayton** [comment]
Rebase please

**Comment by AlexNPavel** [comment]
/lgtm

Can you add the new test to the testgrid dashboard?
The test would need to be added to this section: https://github.com/kubernetes/test-infra/blob/master/config/testgrids/openshift/openshift.yaml#L10
And this section: https://github.com/kubernetes/test-infra/blob/master/config/testgrids/openshift/openshift.yaml#L129

**Pull request merged**